### DISCUSSION TITLE
Time of document upload.

### DISCUSSION THREAD

**Discussion by Pranav-NG** [milvus-io/milvus#30041] [Q&A and General discussion]
Hi . .
I am have a set of documents in Milvus that needs to be updated regularly with time.
For that , I wanted to know , is there a way to get the timestamp of each document upload.

Thanks ,

**Comment by yhmo** [comment]
There is a timestamp for each row, but it is hidden, no interface can get the timestamp.
You can define an INT64 field to store your timestamp.

```
fields=[
    FieldSchema(name="id", dtype=DataType.INT64, is_primary=True, auto_id=True),
    FieldSchema(name="vector", dtype = DataType.FLOAT_VECTOR, dim=dim),
    FieldSchema(name="timestamp", dtype = DataType.INT64),
]

schema = CollectionSchema(fields=fields, enable_dynamic_field=False)
collection = Collection(name=collection_name, schema=schema)

collection.insert({"id": 100, "vector": [.....], "timestamp": int(time.time())})
```

**Comment by xiaofan-luan** [comment]
> There is a timestamp for each row, but it is hidden, no interface can get the timestamp. You can define an INT64 field to store your timestamp.
> 
> ```
> fields=[
>     FieldSchema(name="id", dtype=DataType.INT64, is_primary=True, auto_id=True),
>     FieldSchema(name="vector", dtype = DataType.FLOAT_VECTOR, dim=dim),
>     FieldSchema(name="timestamp", dtype = DataType.INT64),
> ]
> 
> schema = CollectionSchema(fields=fields, enable_dynamic_field=False)
> collection = Collection(name=collection_name, schema=schema)
> 
> collection.insert({"id": 100, "vector": [.....], "timestamp": int(time.time())})
> ```

Should we add a feature to get physical timestamp out, that seems to be a easy fix?

**Reply by yhmo** [reply]
The value of the internal timestamp is a hybrid value combined by timestamp and a logic value. Users don't know how to parse it.
